Enterprise-CMCS/macpro-mdct-core
# Adding instructions for development workflow with another product

### Description
This PR is to make modifications to the README to add instructions for a development workflow for mdct-core and consuming it within another one of your applications 

The approach taken uses yarn link as the mdct products all use yarn and it provided the most simple straight forward path. 

In addition the PR adds some language to the README to provide more clarity on releasing to NPM.


### Related ticket(s)
https://jiraent.cms.gov/browse/CMDCT-2937

---
### How to test
Here are steps I used to establish this workflow utilizing the consume-core-test branch in MCR:

1) Deploy MCR locally reflecting no changes.
2) Modify “/services/ui-src/package.json” to import “@enterprise-cmcs/mdct-core”
3) Modified “/services/ui-src/src/components/logins/LoginCognito.tsx” to import reports from core 
4) Deploy the app to see changes from core deployed (consuming from npm)
5) return to core repo and modify “src/types/reports.ts” to make some local changes
6) at the top level of core run yarn link
7) go back to macpro-mdct-mcr/services/ui-src and run yarn link “@enterprise-cmcs/mdct-core”
8) go back to core and run yarn install and yarn run build 
9) go back to mcr and run ./dev local to deploy and see the changes reflected from core in your application without needing to publish out to npm from core 
10) to iterate changes between core and an app you just have to run yarn install and yarn run build in core and redeploy the app 
11) After completed developing between core and your application go to the core repo and run yarn unlink
12) redeploy your application locally to confirm no local core changes are reflected 


### Important updates
none


---
### Author checklist


---
